unitOneAssessmentPractice.js

```diff
@@ -4,74 +4,156 @@ let assert = require('assert')
 
 // Write a function called double that doubles a number
 
+const double = (num) => num * 2
+
+
 // Uncomment out the next line to test your solution
-// runQ1Tests()
+runQ1Tests()
 
 // Question Two:
 
 // Write a function called containsSeven that returns whether or not a number has an sevens in it.
 
+const containsSeven = num => {
+  let numString = num.toString()
+  for (let el of numString){
+    if (el === "7"){
+      return true
+    }
+  }
+  return false
+}
+
+
 // Uncomment out the next line to test your solution
-// runQ2Tests()
+runQ2Tests()
 
 // Question Three:
 
 // Write a function called capitalizeTheAs the capitalizes all of the lowercase "a"s in a string leaving all other characters the same
 
+const capitalizeTheAs = str => {
+  let string = ""  
+  for (let el of str){
+      if (el === "a" ) {
+        string += el.toUpperCase()
+      } else {
+        string += el
+      }
+  }
+  return string
+}
+
+
 // Uncomment out the next line to test your solution
-// runQ3Tests()
+runQ3Tests()
 
 // Question Four:
 // Write a function called largest that returns the largest value in an array
 
+const largest = arr => {
+  arr.sort(function(a,b){return b-a})
+  return arr[0]
+}
+
 // Uncomment out the next line to test your solution
-// runQ4Tests()
+runQ4Tests()
 
 // Question Five:
 // Write a function called average that returns the average of an array
 
+const average = arr => arr.reduce((a,b) => a + b, 0) / arr.length
+  
 // Uncomment out the next line to test your solution
-// runQ5Tests()
+runQ5Tests()
 
 // Question Six:
 // Write a function called mode that returns the most frequently occurring number in an array
 // HINT: Use an object where the keys are the numbers, and the values are how many times they appear in the array.
 
+const mode = arr => {
+    let object = {}
+    for (let num of arr){
+      if (object[num]){
+        object[num] += 1
+      } else (object[num] = 1)
+    }
+  
+    let maxVal = 0
+    let maxKey = ""
+    for (let key in object){
+      if (object[key] > maxVal){
+        maxVal = object[key]
+        maxKey = key
+      } 
+    }
+    return Number(maxKey)
+}
+
 // Uncomment out the next line to test your solution
-// runQ6Tests()
+runQ6Tests()
 
 // Question Seven: (BONUS)
 // Write a function called median that returns the most median number in an array
 // HINT: You'll need to sort the array first
 // This one's also a bit tricky, feel free to skip it and come back to it.
 
+const median = arr => {
+  arr.sort(function(a,b){return b-a})
+  if (arr.length%2 === 0){
+    return (arr[arr.length-2] + arr[arr.length-1])/2
```

Comment:
Here you are returning an average of the last two elements of the array. What you could do is 
 return (arr[Math.floor(arr.length / 2 - 1)] + arr[Math.floor(arr.length/2)])/ 2;